ims/einstein/iscsi/tgt.py

```diff
@@ -11,180 +12,154 @@
 
     @log
     def show_status(self):
-        # arglist = self.arglist
-        # arglist.append('status')
+        """
+        Returns the status of tgtd
+
+        :return: Running or Dead or Error as String
+        """
         command = "service tgtd status"
-        proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-        status_string = proc.communicate(self.root_password + "\n")[0]
+        status_string = shell.call(command, sudo=True)
+        logger.debug("Output = %s", status_string)
         if 'active (running)' in status_string:
             return 'Running'
         elif 'inactive (dead)' in status_string:
             return 'Dead'
         # Have to check if there are any other states
         else:
-            return "Running in error state"
+            return "Error"
 
     def __generate_config_file(self, target_name):
-        config = open(self.TGT_ISCSI_CONFIG + target_name + ".conf", 'w')
+        """
+        Generates the config file in conf.d/
+
+        :param target_name: Target for which config file should be created
+        :return: None
+        """
+        config = open(
+            os.path.join(self.TGT_ISCSI_CONFIG, target_name + ".conf"), 'w')
         template_loc = os.path.abspath(
             os.path.join(os.path.dirname(os.path.abspath(__file__)), "..",
                          ".."))
-        for line in open(template_loc + "/tgt_target.temp", 'r'):
+        for line in open(os.path.join(template_loc, "tgt_target.temp"), 'r'):
             line = line.replace('${target_name}', target_name)
             line = line.replace('${ceph_user}', self.fs_user)
             line = line.replace('${ceph_config}', self.fs_config_loc)
+            line = line.replace('${pool}', self.fs_pool)
             config.write(line)
         config.close()
 
     @log
     def add_target(self, target_name):
-        '''
+        """
         Adds target
-        :param target_name: Name of target to be added.
-        :return:
-        '''
+
+        :param target_name: Name of target to be added
+        :return: None
+        """
         try:
             targets = self.list_targets()
             if target_name not in targets:
                 self.__generate_config_file(target_name)
-                # tgtarglist = ["sudo", "-S", "tgt-admin", "--execute"]
-                command = "tgt-admin --execute".format(self.root_password)
-                proc = subprocess.Popen(command, shell=True,
-                                        stdout=subprocess.PIPE,
-                                        stderr=subprocess.PIPE)
-                proc.communicate()
-                if proc.returncode != 0:
-                    raise iscsi_exceptions.UpdateConfigFailedException(
-                        "Adding new target failed at creating target file \
-                        stage")
+                command = "tgt-admin --execute"
```

Comment:
The ``--execute`` part of the command doesn't need to be in the config file, here it's fine. The executable to run the command on though (eg. tgt-admin), must be configurable.